gpm_api/io/checks.py

```diff
@@ -157,15 +157,15 @@ def check_time(time):
     if isinstance(time, np.ndarray):
         if np.issubdtype(time.dtype, np.datetime64):
             if time.size == 1:
-                time = time.astype("datetime64[s]").tolist()
+                time = time.astype("datetime64[us]").tolist()
             else:
                 raise ValueError("Expecting a single timestep!")
         else:
             raise ValueError("The numpy array does not have a np.datetime64 dtype!")
 
-    # If np.datetime64, convert to datetime.datetime
+    # If np.datetime64, convert to datetime.datetime with microsecond precision
     if isinstance(time, np.datetime64):
-        time = time.astype("datetime64[s]").tolist()
+        time = time.astype("datetime64[us]").tolist()
```

Comment:
Why this change? 
We aim for the output to be a `datetime.datetime` object with second accuracy.

Reply:
@evanjt

Reply:
Apologies, there was a misinterpretation of what was requested in a previous discussion. If I can remember correctly, using ms would round to microseconds, so this change was made, but now I see that it is incorrect. I have reverted it